src/js/pouchdb-node.js

```diff
@@ -0,0 +1,169 @@
+/*
+
+    A node-specific extension of gpii.pouch that adds additional abilities, such as loading data from package-specific
+    paths.
+
+ */
+/* eslint-env node */
+"use strict";
+var fluid = require("infusion");
+var gpii  = fluid.registerNamespace("gpii");
+
+var fs   = require("fs");
+var os   = require("os");
+var path = require("path");
+require("../../");
+
+fluid.registerNamespace("gpii.pouch.node");
+
+/**
+ *
+ * Create our working directory if it doesn't already exist.
+ *
+ * @param that - The component itself.
+ *
+ */
+gpii.pouch.node.initDir = function (that) {
+    var fullPath = path.resolve(that.options.baseDir, that.options.dbOptions.name);
+
+    // fluid.log("pouch component '" , that.id, "' saving data to '", fullPath, "'...");
+
+    if (fs.existsSync(that.options.baseDir)) {
+        if (fs.existsSync(fullPath)) {
+            that.hasExistingDataDir = true;
+        }
+        else {
+            that.hasExistingDataDir = false;
+            fs.mkdirSync(fullPath);
+        }
+    }
+    else {
+        that.hasExistingDataDir = false;
+        fs.mkdirSync(that.options.baseDir);
+    }
+};
+
+gpii.pouch.node.makeSafePrefix = function (toResolve) {
+    return fluid.module.resolvePath(toResolve) + path.sep;
+};
+
+/**
+ *
+ * Expand a single database definition to handle various "short" and "long" formats, including:
+ *
+ *   1. "singlePath" // One single file to be loaded.
+ *   2. ["path1", "path2"] // Multiple files to be loaded.
+ *   3. undefined // No data.
+ *
+ * @param dbPath - A `String` or `Array` representing one or more files to load (see above for the supported variations).
+ * @returns {Array} An array of database paths.
+ *
+ */
+gpii.pouch.node.expandPath = function (dbPath) {
+    if (Array.isArray(dbPath)) {
+        return dbPath;
+    }
+    else if (typeof dbPath === "string") {
+        return [dbPath];
+    }
+
+    // If we were passed something odd, like undefined or null, return an empty array.
+    return [];
+};
+
+/**
+ *
+ * Load data from one or more package-relative paths..
+ *
+ * @param that - The component itself.
+ * @param dbPath - A string or array representing data to be loaded.
+ * @returns {Promise} - A promise which will be resolved when all data has been loaded.
+ *
+ */
+gpii.pouch.node.loadDataFromPath = function (that, dbPath) {
+    var promises = [];
+
+    var expandedPaths = gpii.pouch.node.expandPath(dbPath);
+    fluid.each(expandedPaths, function (dbPath) {
+        if (dbPath) {
```

Comment:
There might simply be no data to load, as is the case with the "noData" example in the tests, but I believe those are screened out in the code that calls loadDataFromPath.  Will confirm.